lib/fluent/plugin/out_bigquery.rb

```diff
@@ -2,374 +2,303 @@
+      # config_param :rows_per_second_limit, :integer, default: 1000 # spike limit
+      ### method: ''Streaming data inserts support
+      #  https://developers.google.com/bigquery/streaming-data-into-bigquery#usecases
+      # Maximum row size: 100 KB
+      # Maximum data size of all rows, per insert: 1 MB
+      # Maximum rows per second: 100 rows per second, per table, with allowed and occasional bursts of up to 1,000 rows per second.
+      #                          If you exceed 100 rows per second for an extended period of time, throttling might occur.
+      ### Toooooooooooooo short/small per inserts and row!
+
+      ## Timeout
+      # request_timeout_sec
+      #   Bigquery API response timeout
+      # request_open_timeout_sec
+      #   Bigquery API connection, and request timeout
+      config_param :request_timeout_sec, :time, default: nil
+      config_param :request_open_timeout_sec, :time, default: 60
+
+      ### Table types
+      # https://developers.google.com/bigquery/docs/tables
+      #
+      # type - The following data types are supported; see Data Formats for details on each data type:
+      # STRING
+      # INTEGER
+      # FLOAT
+      # BOOLEAN
+      # RECORD A JSON object, used when importing nested records. This type is only available when using JSON source files.
+      #
+      # mode - Whether a field can be null. The following values are supported:
+      # NULLABLE - The cell can be null.
+      # REQUIRED - The cell cannot be null.
+      # REPEATED - Zero or more repeated simple or nested subfields. This mode is only supported when using JSON source files.
+
+      def initialize
+        super
+        require 'multi_json'
+        require 'google/apis/bigquery_v2'
+        require 'googleauth'
+        require 'active_support/json'
+        require 'active_support/core_ext/hash'
+        require 'active_support/core_ext/object/json'
+
+        # MEMO: signet-0.6.1 depend on Farady.default_connection
+        Faraday.default_connection.options.timeout = 60
       end
 
-      case @auth_method
-      when :private_key
-        unless @email && @private_key_path
-          raise Fluent::ConfigError, "'email' and 'private_key_path' must be specified if auth_method == 'private_key'"
+      def configure(conf)
+        if conf["method"] == "load"
+          configure_for_load(conf)
+        else
+          configure_for_insert(conf)
         end
-      when :compute_engine
-        # Do nothing
-      when :json_key
-        unless @json_key
-          raise Fluent::ConfigError, "'json_key' must be specified if auth_method == 'json_key'"
+        super
+
+        case @method
+        when :insert
+          extend(InsertImplementation)
+        when :load
+          raise Fluent::ConfigError, "'template_suffix' is for only `insert` mode, instead use 'fetch_schema_table' and formatted table name" if @template_suffix
+          extend(LoadImplementation)
+        else
```

Comment:
The code can't reach this `else` actually... Fluentd raises configuration error if users specify a value not in `list`.